pkg/lib/cockpit/location.ts

```diff
@@ -0,0 +1,192 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+import { url_root, calculate_application } from './_internal/location-utils';
+
+/* HACK: Mozilla will unescape 'window.location.hash' before returning
+* it, which is broken.
+*
+* https://bugzilla.mozilla.org/show_bug.cgi?id=135309
+*/
+export function get_window_location_hash() {
```

Comment:
tsc may divine the return code here, but I'd appreciate if we could keep being explicit (like python mypy). I.e. add `: string` return annotation here.